### PR TITLE
Add ARKV6X

### PR DESCRIPTION
**Pull request by AlexKlimaj**
This PR adds support for the ARKV6X flight controller. It is based on the Pixhawk Autopilot Bus form factor and FMUV6X pinout. There are differences in sensors with the FMUV6X, enough to warrant its own hwdef. There is also a timer connected to all the IMU's to provide an external clock.

https://arkelectron.com/product/arkv6x/

Current Testing Status
- [X] Builds
- [X] Communicates with QGC daily
- [X] Flashes through QGC
- [X] All sensors present
- [X] IMU orientations correct
- [x] PWM outputs
- [x] RC input
- [x] IO MCU communication
- [x] Digital power module communication - Need to set BATT_I2C_BUS to 1 or 2. Are both supported at the same time?
- [x] Telem ports
- [x] GPS ports
- [x] Buzzer
- [x] CAN ports
- [x] I2C ports
- [x] Flight tested

[00000012.zip](https://github.com/ArduPilot/ardupilot/files/9803837/00000012.zip)
[00000013.zip](https://github.com/ArduPilot/ardupilot/files/9803899/00000013.zip)
[00000015.zip](https://github.com/ArduPilot/ardupilot/files/9804075/00000015.zip)

### CODE REVIEW

**Comment by AlexKlimaj** [comment]
It looks like two of the same IMU part number on different SPI buses isn't supported in Ardupilot?

**Comment by AlexKlimaj** [comment]
I have it flashed with just 2/3 IMUs. Looks like the sensor orientations are different than PX4? I am using the same rotations in the startup but my horizon is upside down and pitch/roll are reversed.

File: libraries/AP_Baro/AP_Baro_BMP388.cpp [hunk 1]
```diff
@@ -91,8 +93,10 @@ bool AP_Baro_BMP388::init()
     uint8_t whoami;
     if (!read_registers(BMP388_REG_ID, &whoami, 1)  ||
         whoami != BMP388_ID) {
-        // not a BMP388
-        return false;
+        if (whoami != BMP390_ID) {
```
Comment (tridge) [inline libraries/AP_Baro/AP_Baro_BMP388.cpp:96]: the nested if is a bit odd, I'd prefer a switch statement, and create a new DEVTYPE_BARO_BMP390 and use that in the dev->set_device_type() call. That way in a log we can see what sort of baro we have (it shows up in the "devid" command in MAVProxy for example, and in MAVExplorer)

File: libraries/AP_HAL_ChibiOS/hwdef/ARKV6X/hwdef.dat [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+# crystal frequency
+OSCILLATOR_HZ 16000000
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 2
+
+# board ID for firmware load
+APJ_BOARD_ID 57
+
+# enable board sub-type detection for FMUV6 TODO: Figure out if this is needed for ARKV6X
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/ARKV6X/hwdef.dat:16]: its not needed as you use a separate board ID. This is just used to distinguish the Holybro and CUAV 6X variants

**Review by tridge — CHANGES_REQUESTED** [review]
looks like a nice board! Just needs a small baro change

File: libraries/AP_Baro/AP_Baro_BMP388.cpp (hunk 1 above)
Comment (AlexKlimaj) [reply libraries/AP_Baro/AP_Baro_BMP388.cpp]: @tridge Okay I think I have all of your comments addressed and the commits renamed.

**Comment by tridge** [comment]
i've fixed up the conflicts and force pushed

**Review by tridge — APPROVED** [review]

**Comment by AlexKlimaj** [comment]
Thanks @tridge !

**Pull request merged**